python/mxnet/gluon/trainer.py

```diff
@@ -109,7 +115,12 @@ def _init_kvstore(self):
         arg_arrays = {param.name: param.data(self._contexts[0]) for param in self._params}
         kvstore, update_on_kvstore = _create_kvstore(self._kvstore, len(self._contexts),
                                                      arg_arrays)
+        update_on_kvstore = self._update_on_kvstore if self._update_on_kvstore is not None \
+                            else update_on_kvstore
         if kvstore:
+            if kvstore.type != 'local' and not self._update_on_kvstore:
+                warnings.warn('Turning off update_on_kvstore in Trainer may cause updates '
+                              'to be inefficient for "{}" kvstore.'.format(kvstore.type))
```

Comment:
The warning is confusing. update_on_kvstore is actually set to False for dist kvstore in line 127..

Reply:
I don't think we need this warning. If user set update_on_kvstore explicitly we should assume they have read the doc for that option